nextflow.config

```diff
@@ -1,154 +1,287 @@
 /*
- * -------------------------------------------------
- *  nf-core/circrna Nextflow config file
- * -------------------------------------------------
- * Default config options for all environments.
- */
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    nf-core/circrna Nextflow config file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Default config options for all compute environments
+----------------------------------------------------------------------------------------
+*/
 
 // Global default params, used in configs
+// TODO: tidy up after workflow is running smoothly
 params {
 
-  // Workflow flags
-  // TODO nf-core: Specify your pipeline's command line flags
-  genome = false
-  input = "data/*{1,2}.fastq.gz"
-  single_end = false
-  outdir = './results'
-  publish_dir_mode = 'copy'
-
-  // Boilerplate options
-  name = false
-  multiqc_config = false
-  email = false
-  email_on_fail = false
-  max_multiqc_email_size = 25.MB
-  plaintext_email = false
-  monochrome_logs = false
-  help = false
-  igenomes_base = 's3://ngi-igenomes/igenomes/'
-  tracedir = "${params.outdir}/pipeline_info"
-  igenomes_ignore = false
-  custom_config_version = 'master'
-  custom_config_base = "https://raw.githubusercontent.com/nf-core/configs/${params.custom_config_version}"
-  hostnames = false
-  config_profile_description = false
-  config_profile_contact = false
-  config_profile_url = false
-
-  // Defaults only, expecting to be overwritten
-  max_memory = 128.GB
-  max_cpus = 16
-  max_time = 240.h
+    // Input options
+    input                      = null
+    outdir                     = './results'
+    phenotype                  = null
 
-}
+    // workflow options
+    module                     = 'circrna_discovery'
+    tool                       = 'circexplorer2'
+    bsj_reads                  = 0
+    tool_filter                = 0
+    duplicates_fun             = 'mean'
+    exon_boundary              = 200
+    save_intermediates         = true
+
+    // reference genome options
+    genome                     = null
+    igenomes_base              = 's3://ngi-igenomes/igenomes'
+    igenomes_ignore            = false
+    bowtie                     = null
+    bowtie2                    = null
+    bwa                        = null
+    hisat2                     = null
+    hisat2_build_memory        = '200.GB'
```

Comment:
Should this be guided by `modules.conf` and `base.config` etc?

Reply:
yep imo all ressource requests should be made form the base.config